test/tests/CMakeLists.txt

```diff
@@ -0,0 +1,60 @@
+cmake_minimum_required(VERSION 2.6)
+
+# Bootstrap GoogleTest
+INCLUDE(ExternalProject)
+
+ExternalProject_Add(
+		googletest-distribution
+		URL https://github.com/google/googletest/archive/release-1.8.0.tar.gz
+		URL_HASH SHA1=e7e646a6204638fe8e87e165292b8dd9cd4c36ed
+		TIMEOUT 10
+		CONFIGURE_COMMAND ""
+		BUILD_COMMAND ""
+		INSTALL_COMMAND ""
+)
+
+# Specify include dir
+ExternalProject_Get_Property(googletest-distribution SOURCE_DIR)
+set(GOOGLETEST_DISTRIB_SOURCE_DIR "${SOURCE_DIR}")
+
+ExternalProject_Add(
+		googletest
+		DEPENDS googletest-distribution
+		DOWNLOAD_COMMAND ""
+		SOURCE_DIR "${GOOGLETEST_DISTRIB_SOURCE_DIR}/googletest"
+		# Disable install step
+		INSTALL_COMMAND ""
+		# Wrap download, configure and build steps in a script to log output
+		LOG_DOWNLOAD ON
+		LOG_CONFIGURE ON
+		LOG_BUILD ON)
+
+
+# Specify include dir
+set(GTEST_INCLUDE_DIR ${GOOGLETEST_DISTRIB_SOURCE_DIR}/googletest/include)
+
+# Library
+ExternalProject_Get_Property(googletest BINARY_DIR)
+set(GOOGLETEST_BINARY_DIR "${BINARY_DIR}")
+set(GTEST_LIBRARY_PATH ${GOOGLETEST_BINARY_DIR}/${CMAKE_FIND_LIBRARY_PREFIXES}gtest.a)
```

Comment:
I can even make it `${CMAKE_STATIC_LIBRARY_PREFIX}${GTEST_LIBRARY}${CMAKE_STATIC_LIBRARY_SUFFIX}` :P